### PR TITLE
Add rewards modal

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Created the rewards modal (appears upon clicking "Apply Rewards")
- Displays "Apply $5 Reward" or "$5 Reward Applied"
- Greyed out if total price < 5 and no rewards are currently applied (this is the only case in which we cannot apply rewards; otherwise a reward is currently applied and may be removed)
- Changed everything to arrow notation for functions

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources
Simple modals: https://reactnative.dev/docs/modal

Child-parent data communication:
https://medium.com/@ruthmpardee/passing-data-between-react-components-103ad82ebd17
https://github.com/ryanflorence/react-training/blob/gh-pages/lessons/03-simple-component-communication.md

Not-so-simple modals: https://blog.brainsandbeards.com/better-modals-in-react-native-8ea6fb207146 (did not implement this)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
[Design] Probably needs to show "total price" in the reward modal somewhere, since it's updating as rewards are being applied, and you can't apply a reward if total price is <= $5.
- Probably also want to consider whether we should actually display rewards that can't be applied🤔
- Note that greying out / changing the status of the rewards to be an error state would be a lot more annoying to implement.

[Practice] Mutating state in setState: should use this https://github.com/kolodny/immutability-helper (cart, status in modals, etc)

- Couldn't figure it out exactly and didn't want to waste more time. But it's bad practice to mutate state directly. For attributes, just use the `(prevState) =>` callback; less obvious with arrays/items.
- https://stackoverflow.com/questions/29537299/react-how-to-update-state-item1-in-state-using-setstate

- Various reasons why we should:
https://stackoverflow.com/questions/42037369/how-to-edit-an-item-in-a-state-array/42037439
https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/

[Platform support] TouchableHighlight vs. TouchableOpacity:
- Maybe something we should be tracking:
https://medium.com/differential/better-cross-platform-react-native-components-cb8aadeba472

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

Greyed out 
image

Available
image

Modal pop-up
image

Modal - applied
image

Applied but total is less than 5 (can remove)
image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/CheckoutScreen.js [hunk 1]
```diff
@@ -231,14 +191,14 @@ export default class CheckoutScreen extends React.Component {
...
             </View>
-            <View style={{ height: '20%', paddingBottom: '5%' }}>
-              <TextHeader>Rewards Applied</TextHeader>
-              <ScrollView style={{ alignSelf: 'flex-end' }}>{this.generateRewardsApplied()}</ScrollView>
-            </View>
-            <View style={{ height: '20%', paddingBottom: '5%' }}>
-              <TextHeader>Rewards Available</TextHeader>
-              <ScrollView style={{ alignSelf: 'flex-end' }}>{this.generateRewardsAvailable()}</ScrollView>
-            </View>
+            {/* Should be greyed out if totalPrice < 5 */}
```
Comment (thumn) [inline screens/CheckoutScreen.js]: I think we talked about this earlier but will we ever have a situation where customers want to use a reward on a transaction less than $5 so they don't have to pay any money? Probs not in scope for this PR but something to think about...we'd also have to consider what this means in terms of logic for having the button grayed out (i.e. changing it to a clickable button if a user has rewards in their account at all vs only having it be clickable when they have rewards and the transaction > 5)

Comment (thumn) [reply screens/CheckoutScreen.js]: @12aschen another design question for the future xd

Comment (anniero98) [reply screens/CheckoutScreen.js]: Have added this to quacestions doc!

File: screens/ClerkLoginScreen.js [hunk 2]
```diff
@@ -48,6 +48,7 @@ export default class ClerkLoginScreen extends React.Component {
   _asyncLoginClerk = async clerkRecord => {
     await AsyncStorage.setItem('clerkId', clerkRecord.id);
     await AsyncStorage.setItem('storeId', this.state.storeId);
+    this.setState({ errorMsg: null, password: '' });
```
Comment (anniero98) [inline screens/ClerkLoginScreen.js]: This was just a bug I caught 😅 For login

**Comment by anniero98** [comment]
Closing this PR in favor of #14 - I made some styling updates that I unfortunately couldn't separate out while working on the quantity modal. Regardless, the functionality is 100% in #14, so it's safe to merge #14 then update the rewards modal according to the new design in a new PR.
